src/main/java/org/circle8/dao/RecorridoDao.java

```diff
@@ -246,6 +279,60 @@ public void delete(Transaction t, long id, long zonaId) throws PersistenceExcept
 		}
 	}
 
+
+	public void updateDate(Transaction t, long id, RecorridoEnum o) throws NotFoundException, PersistenceException {
+		String updateSql;
+		if(o == RecorridoEnum.FIN)
+			updateSql = UPDATE_FIN;
+		else
+			updateSql = UPDATE_INICIO;
+		try ( var update = t.prepareStatement(updateSql) ) {
+			update.setTimestamp(1, Timestamp.from(ZonedDateTime.now(Dates.UTC).toInstant()));
+			update.setLong(2, id);
+
+			if(update.executeUpdate() <= 0){
+				throw new NotFoundException("updating the recorrido failed, no affected rows");
+			}
+
+		} catch (SQLException e) {
+			throw new PersistenceException("error updating recorrido", e);
+		}
+	}
+	public void putSave(Transaction t, Recorrido r) throws PersistenceException, NotFoundException {
+		try (var put = createPut(t, r)){
+			val s = put.toString();
+			if(put.executeUpdate() <= 0){
+				throw new NotFoundException("updating the recorrido failed, no affected rows");
+			}
+		} catch (SQLException e) {
+			if(e.getMessage().contains("Recorrido_RecicladorId_fkey"))
+				throw new PersistenceException("El recicladorId ingresado no existe en la tabla", e);
+			throw new PersistenceException("error updating recorrido", e);
+      }
+   }
+
+	private PreparedStatement createPut(Transaction t, Recorrido r) throws PersistenceException, SQLException {
+		val put = new StringBuilder();
+		List<Object> parameters = new ArrayList<>();
+		if(r.recicladorId != null){
+			put.append(SET_RECICLADOR);
+			parameters.add(r.recicladorId);
+		}
+		if(r.fechaRetiro != null){
+			if(!put.isEmpty())
+				put.append(", ");
+			put.append(SET_FECHA_RETIRO);
+			parameters.add(r.fechaRetiro);
+		}
+		parameters.add(r.id);
+		parameters.add(r.zonaId);
```

Comment:
Una mejora, para evitar los `put.isEmpty()` en el caso de agregar nuevos `SET x = ?`.

```suggestion
		val sets = new ArrayList<String>();
		List<Object> parameters = new ArrayList<>();
		if(r.recicladorId != null){
			sets.add(SET_RECICLADOR);
			parameters.add(r.recicladorId);
		}
		if(r.fechaRetiro != null){
			set.add(SET_FECHA_RETIRO);
			parameters.add(r.fechaRetiro);
		}
		parameters.add(r.id);
		parameters.add(r.zonaId);
		val put = String.join(", ", sets);
```

Reply:
Me gusta, hecho.